cmake/HandleGTest.cmake

```diff
@@ -76,7 +76,12 @@ macro(build_external_gtest)
 endmacro(build_external_gtest)
 
 if (BENCHMARK_ENABLE_GTEST_TESTS)
-  if (IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/googletest)
+  if (TARGET gtest AND TARGET gmock AND TARGET gmock_main)
+    # If gtest targets are already defined, use them.
+    # This allows a project to include both googletest and benchmark
+    # as top-level git submodule.
+    set(GTEST_BOTH_LIBRARIES gtest gmock gmock_main)
+  elseif (IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/googletest)
```

Comment:
What should happen when you've explicitly checked out `googletest` into `benchmark/`, but also have already defined the targets? It's not clear what version we should use.

Also, what happens the second time this CMake file is processed? AKA, the first pass we create the targets ourselves. The second pass we detect that the targets are present?

Reply:
If gtest/gmock targets are already defined, we should not use `benchmark/googletest`. Otherwise CMake complains about duplicated targets, as it requires target names being globally unique [https://cmake.org/cmake/help/v3.9/policy/CMP0002.html].

```text
CMake Error at benchmark/googletest/googletest/cmake/internal_utils.cmake:145 (add_library):
  add_library cannot create target "gtest" because another target with the
  same name already exists.  The existing target is a static library created
  in source directory
  "/home/schen/git/recipes/benchmark/googletest/googletest".
  See documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  benchmark/googletest/googletest/cmake/internal_utils.cmake:197 (cxx_library_with_type)
  benchmark/googletest/googletest/CMakeLists.txt:125 (cxx_library)
```

I don't quite understand how this file could be processed twice.  It's only included by `root/CMakeLists.txt`. If `root/CMakeLists.txt` is added twice, I am pretty sure we will have the same duplicate targets error as above.